gym/lib/gym/code_signing_mapping.rb

```diff
@@ -107,23 +107,45 @@ def detect_project_profile_mapping
       self.project_paths.each do |project_path|
         UI.verbose("Parsing project file '#{project_path}' to find selected provisioning profiles")
 
+        # Storing bundle identifiers with duplicate profiles
+        # for informing user later on
+        bundle_identifiers_with_duplicates = []
```

Comment:
One thing I noticed is that we set this to `[]` here, but we check if it's empty inside the `begin` and the loop? is that correct? How does it behave with multiple targets?